test/rabbit_trust_store_test.erl

```diff
@@ -309,6 +310,76 @@ whitelist_directory_DELTA_test_() ->
        end
       }]}.
 
+replaced_whitelisted_certificate_should_be_accepted_test_() ->
+
+    {setup,
+     fun() ->
+             ok = force_delete_entire_directory(friendlies()),
+             ok = build_directory_tree(friendlies())
+     end,
+     fun(_) ->
+             ok = force_delete_entire_directory(friendlies())
+     end,
+    [{timeout,
+      15,
+      fun () ->
+
+             %% Given: a root certificate and a 2 other certificates
+             {Root, Cert, Key}      = ct_helper:make_certs(),
+             {_,  CertFirst, KeyFirst}    = ct_helper:make_certs(),
+             {_,  CertUpdated, KeyUpdated}    = ct_helper:make_certs(),
+
+             ok = rabbit_networking:start_ssl_listener(port(), [{cacerts, [Root]},
+                                                                {cert, Cert},
+                                                                {key, Key} | cfg()], 1),
+
+             ok = whitelist(friendlies(), "bart", CertFirst,  KeyFirst),
+             ok = change_configuration(rabbitmq_trust_store, [{directory, friendlies()},
+                                                              {refresh_interval, {seconds, interval()}}]),
+
+             wait_for_trust_store_refresh(),
+
+             %% Then: a connection using the  
+             {ok, Con} =
+                 amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                            port = port(),
+                                                            ssl_options = [{cert, CertFirst},
+                                                                           {reuse_sessions, false},
```

Comment:
It's worth explaining why you had to do this with a comment. FTR: it was explained to the team on Slack.